### ISSUE TITLE
Scarlett 4i4 4th gen silence

### ISSUE THREAD

**Issue by psilord** [geoffreybennett/alsa-scarlett-gui#142]
# `alsa-scarlett-gui` Issue Template

Thank you for taking the time to contribute to the `alsa-scarlett-gui` project. Before you submit your issue, please ensure you have checked the FAQ and provide the necessary information below.

## Confirmation
- [ x] I confirm that I have read the [FAQ](https://github.com/geoffreybennett/alsa-scarlett-gui/blob/master/FAQ.md).

## Issue Category
Please select the category that best describes your issue:
- [ x] Help Request
- [ ] Bug Report
- [ ] Feature Request

## Environment Details
Please provide the following details about your environment.

### Linux Distribution and Version
(paste output from `cat /etc/redhat-release` or `cat /etc/lsb_release` here)
- Distribution: Ubuntu
- Version: 24.04.1 LTS

### Kernel Version
(paste output from `uname -r` here)
- Kernel version: 6.8.0-45-generic

### Kernel Messages
(paste output from `dmesg | grep -A 5 -B 5 -i focusrite` here)
```text
[    1.389848] clocksource: tsc: mask: 0xffffffffffffffff max_cycles: 0x2e02ca5091a, max_idle_ns: 440795219104 ns
[    1.389940] clocksource: Switched to clocksource tsc
[    1.464721] usb 1-5: New USB device found, idVendor=1235, idProduct=821a, bcdDevice= 8.3c
[    1.464750] usb 1-5: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[    1.464757] usb 1-5: Product: Scarlett 4i4 4th Gen
[    1.464763] usb 1-5: Manufacturer: Focusrite
[    1.464768] usb 1-5: SerialNumber: XXXXXXXXXXXXXX
[    1.578155] usb 2-7: new SuperSpeed USB device number 3 using xhci_hcd
[    1.591712] usb 2-7: New USB device found, idVendor=1058, idProduct=25a3, bcdDevice=10.31
[    1.591727] usb 2-7: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    1.591749] usb 2-7: Product: Elements 25A3
--
[   37.769436] nvidia_uvm: module uses symbols nvUvmInterfaceDisableAccessCntr from proprietary module nvidia, inheriting taint.
[   37.828832] nvidia-uvm: Loaded the UVM driver, major device number 508.
[   37.946053] nvidia-modeset: WARNING: GPU:0: Ancor Communications Inc ROG PG279Q (DP-0): Failed to initialize G-SYNC
[   38.123859] nvidia-modeset: WARNING: GPU:0: Ancor Communications Inc ROG PG279Q (DP-0): Failed to initialize G-SYNC
[   38.208386] nvidia-modeset: WARNING: GPU:0: Ancor Communications Inc ROG PG279Q (DP-0): Failed to initialize G-SYNC
[   38.376093] usb 1-5: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x821a); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[   38.376299] usb 1-5: Firmware version 2108
[   38.378767] usb 1-5: Quirk or no altset; falling back to MIDI 1.0
[   38.379055] usbcore: registered new interface driver snd-usb-audio
[   41.427636] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   41.427640] Bluetooth: BNEP filters: protocol multicast
--
[11327.483746] audit: type=1400 audit(1728030646.599:340): apparmor="DENIED" operation="unlink" class="file" profile="snap.firefox.firefox" name="/dev/char/195:254" pid=31044 comm="Renderer" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
[11538.480687] usb 1-5: new high-speed USB device number 11 using xhci_hcd
[11538.607473] usb 1-5: New USB device found, idVendor=1235, idProduct=821a, bcdDevice= 8.3c
[11538.607505] usb 1-5: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[11538.607513] usb 1-5: Product: Scarlett 4i4 4th Gen
[11538.607520] usb 1-5: Manufacturer: Focusrite
[11538.607526] usb 1-5: SerialNumber: XXXXXXXXXXXXXX
[11540.256527] usb 1-5: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x821a); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[11540.256789] usb 1-5: Firmware version 2108
[11540.262307] usb 1-5: Quirk or no altset; falling back to MIDI 1.0
[11565.288561] usb 1-5: USB disconnect, device number 11
[11568.922032] usb 1-5: new high-speed USB device number 12 using xhci_hcd
[11569.048795] usb 1-5: New USB device found, idVendor=1235, idProduct=821a, bcdDevice= 8.3c
[11569.048814] usb 1-5: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[11569.048822] usb 1-5: Product: Scarlett 4i4 4th Gen
[11569.048829] usb 1-5: Manufacturer: Focusrite
[11569.048835] usb 1-5: SerialNumber: XXXXXXXXXXXXXX
[11570.695617] usb 1-5: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x821a); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[11570.695880] usb 1-5: Firmware version 2108
[11570.702204] usb 1-5: Quirk or no altset; falling back to MIDI 1.0
[11874.408036] audit: type=1400 audit(1728031193.533:341): apparmor="DENIED" operation="unlink" class="file" profile="snap.firefox.firefox" name="/dev/char/195:254" pid=31044 comm="Renderer" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
[12136.921648] audit: type=1400 audit(1728031456.052:342): apparmor="DENIED" operation="unlink" class="file" profile="snap.firefox.firefox" name="/dev/char/195:254" pid=31044 comm="Renderer" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
[12885.654766] audit: type=1400 audit(1728032204.799:343): apparmor="DENIED" operation="unlink" class="file" profile="snap.firefox.firefox" name="/dev/char/195:254" pid=31044 comm="Renderer" requested_mask="d" denied_mask="d" fsuid=1000 ouid=0
--
[72225.600374] usb 1-5: USB disconnect, device number 12
[72403.201102] usb 1-5: new high-speed USB device number 17 using xhci_hcd
[72403.327758] usb 1-5: New USB device found, idVendor=1235, idProduct=821a, bcdDevice= 8.3c
[72403.327776] usb 1-5: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[72403.327784] usb 1-5: Product: Scarlett 4i4 4th Gen
[72403.327791] usb 1-5: Manufacturer: Focusrite
[72403.327797] usb 1-5: SerialNumber: XXXXXXXXXXXXXX
[72404.979236] usb 1-5: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x821a); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[72404.979500] usb 1-5: Firmware version 2108
[72404.985980] usb 1-5: Quirk or no altset; falling back to MIDI 1.0
[74161.058902] logitech-djreceiver 0003:046D:C52B.000E: Long HID++ report bad size (15)
[74329.728760] audit: type=1400 audit(1728093650.046:363): apparmor="DENIED" operation="capable" class="cap" profile="/usr/lib/snapd/snap-confine" pid=140333 comm="snap-confine" capability=12  capname="net_admin"
[74329.728765] audit: type=1400 audit(1728093650.046:364): apparmor="DENIED" operation="capable" class="cap" profile="/usr/lib/snapd/snap-confine" pid=140333 comm="snap-confine" capability=38  capname="perfmon"
```
### Focusrite Interface Series and Model
(maybe shown in kernel messages, or paste output from `lsusb -d1235:` if unsure)
- Series (e.g., Scarlett 2nd/3rd/4th Gen, Clarett USB, Clarett+): Scarlett 4th Gen
- Model (e.g., Solo, 2i2, 4i4, etc.): 4i4
- lsusb: Bus 001 Device 017: ID 1235:821a Focusrite-Novation Scarlett 4i4 4th Gen

### Audio System
(use `ps aux | grep -E "pulseaudio|jackd|pipewire"` to check)
- [ ] PulseAudio
- [ ] JACK
- [ x] PipeWire

```text
username     2684  0.0  0.0 129420 30524 ?        S<sl 00:24   0:26 /usr/bin/pipewire-pulse
username    20860  0.0  0.1 788920 47040 ?        Sl   01:15   0:26 /usr/lib/x86_64-linux-gnu/xfce4/panel/wrapper-2.0 /usr/lib/x86_64-linux-gnu/xfce4/panel/plugins/libpulseaudio-plugin.so 12 16777229 pulseaudio PulseAudio Plugin Adjust the audio volume of the PulseAudio sound system
username   153754  0.0  0.0 119184 14592 ?        S<sl 23:24   0:00 /usr/bin/pipewire
username   153756  0.0  0.0  24004  5760 ?        Ssl  23:24   0:00 /usr/bin/pipewire -c filter-chain.conf
```

## Issue Description
Please provide a detailed description of the issue or feature request, including steps to reproduce (if applicable), expected behavior, and actual behavior:

Hello,

First off, I'd like to thank you for writing the kernel and userspace software
to talk to the Scarlett 4i4 4th gen device. I just bought one of these because
of the apparently excellent support on linux for it and it is my first
focusrite product.

I'd like to use the scarlett 4i4 4th gen on linux for recording voice and/or
the output of an electric guitar/bass whose signal was put through various
guitar pedals.

I'm a linux music neophyte (who dabbled in it successfully some years ago), and
I have understood /just/ enough how to set up jack, qjacktl, hydrogen,
zynaddsubfx, jack-keyboard, ardour, and make them all generally talk to each
other via the graph connection UI in qjacktl (along with various topics like
who is the timebase master, etc).

However, I am ignorant of the details of pipewire, pulseaudio, ALSA, and their
odd compatibility or interoperability layers. I am also ignorant of the low
level details of identifying sound cards, their capabilities, and hardware
features about them--how to tell if something is wrong, etc. In real life, I'm
a UNIX C Systems programmer, but I just never had to really dig into the linux
audio system because it mostly just worked for what I needed. However, I
literally could miss a fader setting or a hardware mute or something which
could be the root cause of my problem. :)

I followed your directions in the USAGE.md for turning off Easy Start mode and
restarting the device.  In general the directions all worked and the GUI popped
up and everything *looked* ok. I could open the various interfaces like the
Levels, Routes, Mixer, etc and it appeared there was meaningful information in
them. After reading the manual and playing with your GUI, I generally
understand how the device works. And the physical device matches the UI you
made and I haven't had any surprises there. 

Kudos to your tooltips you placed everywhere, there were highly significant to
explaining to me how to use the device especially the router.

But, I've run into a couple problems and I'm curious what the solutions are.

The first one:

PROBLEM: After starting alsa-scarlett-gui and setting it to "direct" in the
routing menu, the qjackctl graph interface has only two boxes show up for the
4i4 4th gen--the ALSA MIDI ports. I'm also expecting that anytime the PCM
Inputs or PCM output are connected to anything, it updates in realtime the
qjacktl ports. (And I have no idea how to transmogrify the 4i4's ALSA MIDI port
into a JACK MIDI port in case I wanted to have that configuration.)

EXPECTED BEHAVIOR: I was expecting another 4i4 box to show up with 6 inputs and
6 outputs so I can route audio data to/from the PC (and ardour, etc) to the
device. I believe I get no sound from device inputs because I haven't routed it
anywhere in qjackctl...because there are no 4i4 ports to route!

The second one:

PROBLEM: When I start alsa-scarlett-gui, after a little bit the TU106 sound
device, which is acting as the system output, vanishes from pavucontrol leaving
me with no sound (for applications like firefox). The 4i4 still shows up in
pavucontrol, and if I restart pipewire the TU106 shows up again--but no matter
what I cannot coax any sound on the PC transmitted from the 4i4.

EXPECTED BEHAVIOR: I expect other audio devices to not vanish and other
applications to not lose their ability to produce sound.

Sadly, you may have to help a neophyte's linux music configuration little
bit in order to see if there is a real bug in your 4i4 software or not.

Here is a pile of environmental information about my machine which I hope can
help debug the problem. I will glady supply more upon reqiest.

Linux Distro: `Ubuntu 24.04.1 LTS`
--------------------------------
Technically it is XUbuntu.

```text
Linux ****** 6.8.0-45-generic #45-Ubuntu SMP PREEMPT_DYNAMIC Fri Aug 30 12:02:04 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux
```

`lspci | grep -i audio`
---------------------
```text
00:1f.3 Audio device: Intel Corporation 200 Series PCH HD Audio
01:00.1 Audio device: NVIDIA Corporation TU106 High Definition Audio Controller (rev a1)
```

`lsusb  | grep Focus`
-------------------
```text
Bus 001 Device 017: ID 1235:821a Focusrite-Novation Scarlett 4i4 4th Gen
```

`aplay -l`
--------
```text
**** List of PLAYBACK Hardware Devices ****
card 0: NVidia [HDA NVidia], device 3: HDMI 0 [ROG PG279Q]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 0: NVidia [HDA NVidia], device 9: HDMI 3 [HDMI 3]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: Gen [Scarlett 4i4 4th Gen], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: PCH [HDA Intel PCH], device 0: ALC3861 Analog [ALC3861 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: PCH [HDA Intel PCH], device 3: HDMI 0 [HDMI 0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: PCH [HDA Intel PCH], device 7: HDMI 1 [HDMI 1]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: PCH [HDA Intel PCH], device 8: HDMI 2 [HDMI 2]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

`arecord -l`
----------

```text
**** List of CAPTURE Hardware Devices ****
card 1: Gen [Scarlett 4i4 4th Gen], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 2: PCH [HDA Intel PCH], device 0: ALC3861 Analog [ALC3861 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
```

Actual output device
--------------------

My actual output device is are speakers in my monitor plugging into an
`NVIDIA Corporation TU106 [GeForce RTX 2060 SUPER] (rev a1)`
video card.

4i4 detection in dmesg
----------------------

(With a problem I just noticed, but I don't know what caused it, so far I've
only seen it once and only the FIRST time it was plugged in. The disconnect was
me unplugging it. Later attempts at plugging it in don't emit the 1001 lines.)

```text
[    1.337832] usb 1-5: new high-speed USB device number 3 using xhci_hcd
[    1.464721] usb 1-5: New USB device found, idVendor=1235, idProduct=821a, bcdDevice= 8.3c
[    1.464750] usb 1-5: New USB device strings: Mfr=1, Product=3, SerialNumber=2
[    1.464757] usb 1-5: Product: Scarlett 4i4 4th Gen
[    1.464763] usb 1-5: Manufacturer: Focusrite
[    1.464768] usb 1-5: SerialNumber: ********
[   38.376093] usb 1-5: Focusrite Scarlett Gen 4 Mixer Driver enabled (pid=0x821a); report any issues to https://github.com/geoffreybennett/scarlett-gen2/issues
[   38.376299] usb 1-5: Firmware version 2108
[   38.378767] usb 1-5: Quirk or no altset; falling back to MIDI 1.0
[ 1205.312751] usb 1-5: Scarlett Gen 4 USB response result cmd 1001 was -71 expected 112
[ 1210.657115] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1215.777002] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1220.896871] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1226.016737] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1231.136605] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1236.256538] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1241.376370] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 1246.496310] usb 1-5: Scarlett Gen 4 USB request result cmd 1001 was -110
[ 3138.205488] usb 1-5: USB disconnect, device number 3
```

Group Information
-----------------

I've added my user to the "audio" group and logged out/in. `id` reports that
I am in the audio group.

`/etc/security/limits.d/10-audio.conf` contains
---------------------------------------------
```text
@audio - rtprio 95
@audio - memlock unlimited
@audio  - nice   -19
```

alsa-scarlett-gui
-----------------

I checked out alsa-scarlett-gui and successfully built main:abdb7f.  For this
ticket, assume I've reset the 4i4 routes to "Direct" (after being cleared), the
Inst light is white, the 1 light is green, Sync Status "Locked" is green, Power
External is white, Sample Rate N/A, Link is off, Safe, Inst, Air, 48V, Autogain
is off, it says "Success". The first time I used asla-scarlett-gui and turned
off easy mode, there were no prompts for uploading firmware.

alsa-scarlett physical device
-----------------------------

I have a turned on (no batteries needed) and plugged mic using a 1/4 phono jack
into the Hardware Input Analogue 1 input on the device. The Inst light is
white. The input level is set almost loud enough to turn red (in the UI, not
on the device because it has direct routing) when I forcefully tap the mic. The
device is powered with the 5V supplied power supply and that is plugged into
the correct 5V port. I used the USB cable that came with the device (including
the USB-C -> USB-A converter) and it is plugged directly into a black Usb 2.0
port on the computer (no intermediary hub).  The USB light is green. The 1
light is green (because that is where I plugged in the mic). When I tap the
mic, the light around the level is black, because I used the "direct" setting
which bypasses the DSP.

Audio program starting sequence with actual cmdline args
--------------------------------------------------------

```text
# Herein I reproduce my setup.

# First, install the identified Ubuntu (which is really XUbuntu since I use
# the xfce4 desktop.)

# In these steps, everything is run as my user id (NOT root).

# jackdmp version 1.9.21 tmpdir /dev/shm protocol 9
> jackd -d alsa -d hw:0,3

# QjackCtl 0.9.13
# Qt: 6.4.2
> qjackctl &

# Version: 3.0.6
> zynaddsubfx
	This automatically shows up in the qjackctl graph.
  I connect zynaddsubfx's "out 1" and "out 2" ports to the system ports.
	Then I change the Nio Settings to JACK for MIDI.
	Then the correct jack MIDI-in port shows up in qjackctl.
  I pick a decent instrument.

# jack-keyboard v2.7.2
> jack-keyboard
	This automatically shows up in the qjacktl with a MIDI-out port.
  I connect it to zynaddsubfx's jack midi in port.
  I can use the mouse to play the keyboard and it sound comes out.

# main:abdb7f 
> alsa-scarlett-gui
  The GUI shows up, everything looks like it works.
  Two boxes automatically show up in qjackctl, a "Scarlett 4i4 4th Gen MIDI 1"
  port on the left of one box and in the other box the same name on the right.
```

At this point, jackd keeps working, but the TU106 device vanishes from
pavucontrol. This means things like firefox are now silent, but zynaddsubfx can
still play audible notes! I'm unsure of the state of audio system. jackd keeps
working, but the TU106 device is gone!?!

If I run this:

```text
> systemctl --user restart pipewire.service
```

I can restart the TU106 device and sound works again for things like firefox.
It is unknown if the device later vanishes when the alsa-scarlett-gui program
was already running, but it seems to always vanish if I start the program.

I've tinkered a few hours carefully trying to debug what could be going wrong,
which I assume is just something easy because I am ignorant. But, alas, I can't
find it so here I am, asking for support. :)

Thank you!

-pete

**Comment by geoffreybennett** [comment]
Hi Pete,

Thanks for your questions! I think your main issue is that you are starting JACK, and it will be fixed by not starting JACK :crossed_fingers:

First note that ALSA (i.e. the kernel driver) only supports one application at a time using a sound card, and JACK (by default) only supports using one sound card at time.

So when you run `jackd -d alsa -d hw:0,3`, the TU106 device vanishes from pavucontrol because that's the device you told JACK to use, and PipeWire (which pavucontrol is talking with) doesn't get to use it anymore. Therefore your applications like Firefox don't get sound anymore because JACK took over the device, and your audio applications like qjackctl can't see the Scarlett because you didn't tell JACK to use that device. On the bright side, Firefox should be able to record audio from your Scarlett :)

PipeWire supports using all your sound cards at the same time, and it has JACK and Pulseaudio-compatible interfaces. So I think the solution to your problems is just: don't use JACK.

Use instead:
- `pavucontrol` to select the Pro Audio configuration for your Scarlett
- `pavucontrol` to monitor the input level and check if it's seeing audio from your Scarlett
- `qpwgraph` to monitor what's being connected to your audio devices' I/O ports (it works a lot like the qjackctl graph window)

Then everything should just work.

> I'm also expecting that anytime the PCM Inputs or PCM output are connected to anything, it updates in realtime the qjacktl ports.

No, alsa-scarlett-gui just configures the routing within the Scarlett. You should see those ports in qjackctl or qpwgraph, and think of them as connected via the USB cable:

![image](https://github.com/user-attachments/assets/123a4162-4210-432d-95ad-48c8bc889c05)

What you send inside the Scarlett to the PCM Input 1 will appear in capture_AUX0, etc. Similarly for playback_AUX0 — what you send there will appear inside the Scarlett at PCM Output 1, etc.

> (And I have no idea how to transmogrify the 4i4's ALSA MIDI port into a JACK MIDI port in case I wanted to have that configuration.)

Me either sorry. That's a JACK or PipeWire or ALSA thing, not Scarlett-related.

I do note one other issue, that "Scarlett Gen 4 USB response result cmd 1001 was -71 expected 112" error and subsequent errors are caused by a bug in that particular kernel which Ubuntu is using. If you don't see it again, all good, but it's likely to pop up every so often and stop the driver loading, or stop alsa-scarlett-gui from working. You can build the backported driver from here: https://github.com/geoffreybennett/scarlett-gen2/releases to fix that.

Please let me know how you go, and if you have any other questions!

Regards,
Geoffrey.

**Comment by psilord** [comment]
Hello! After careful reading of your instructions and inspection of my system, I have gotten the Scarlett 4i4 4th Gen to function appropriately! You were definitely right, JACK had to be turned off.

However, some caveats happened, which I will describe here to help you for the next person:

I had to install the pipewire-jack package because  a couple of my audio tools (like Ardour) ONLY speak jack, and this package provides a tool (among other things, like a jack appropriate pipewire interface) called `pw-jack` that you use to start your audio tool, like `pe-jack ardour`. This allows the tool to connect to the jack interface provided by the pipewire-jack package.

I had to compile my own zynaddsubfx because the older one that came with my distro wanted full control of the audio system it was using no matter what and did not play well with others.

Once the pipewire-jack package entered the picture, most things (which only had a JACK option) got straight forward to get working again.

I seriously appreciate your time and answers to my questions. I've now gotten Ardour to record tracks from various instruments and such and things are looking great.

The scarlett 4i4 seems to be working excellently too.

Thank you!
